src/NuGet.Core/NuGet.Commands/RestoreCommand/Logging/RestoreCollectorLogger.cs

```diff
@@ -142,6 +156,23 @@ protected bool DisplayMessage(IRestoreLogMessage message)
             }   
         }
 
+        private void PopulateTransitiveWarningPropertiesCollection(ILogMessage message)
+        {
+            // Populate TransitiveWarningPropertiesCollection only if it is null and we have RestoreTargetGraphs.
+            // This will happen at most once and only if we have the project spec with restore metadata.
+            if (message.Level == LogLevel.Warning &&
+                TransitiveWarningPropertiesCollection == null &&
+                RestoreTargetGraphs != null &&
+                RestoreTargetGraphs.Any() &&
```

Comment:
discussed offline - the confusion was that RestoreTargetGraphs is for the project and not the message. This check is done for cases when a warning is thrown before the project target graphs are not yet generated. 

I will leave a comment to make this clearer.